weaver/wps_restapi/jobs/jobs.py

```diff
@@ -387,33 +390,88 @@ def get_job_status(request):
     return HTTPOk(json=job_status)
 
 
-def cancel_job_task(job, container):
+def raise_job_dismissed(job, container=None):
+    # type: (Job, Optional[AnySettingsContainer]) -> None
+    """
+    Raise the appropriate messages for dismissed job status.
+    """
+    if job.status == status.STATUS_DISMISSED:
+        # provide the job status links since it is still available for reference
+        settings = get_settings(container)
+        job_links = job.links(settings)
+        job_links = [link for link in job_links if link["rel"] in ["status", "alternate", "collection", "up"]]
+        raise JobGone(
+            json={
+                "code": "JobDismissed",
+                "description": "Job was dismissed and artifacts have been removed.",
+                "value": job.id,
+                "links": job_links
+            }
+        )
+
+
+def dismiss_job_task(job, container):
     # type: (Job, AnySettingsContainer) -> Job
-    app.control.revoke(job.task_id, terminate=True)  # signal to stop celery task. Up to it to terminate remote if any.
+    """
+    Cancels any pending or running :mod:`Celery` task and removes completed job artifacts.
+
+    :param job: job to cancel or cleanup.
+    :param container:
+    :return:
+    """
+    raise_job_dismissed(job, container)
+    if job.status in status.JOB_STATUS_CATEGORIES[status.JOB_STATUS_CATEGORY_RUNNING]:
+        # signal to stop celery task. Up to it to terminate remote if any.
+        LOGGER.debug("Job [%s] dismiss operation: Canceling task [%s]", job.id, job.task_id)
+        app.control.revoke(job.task_id, terminate=True)
```

Comment:
Did you take in consideration a persistent revoke : https://docs.celeryproject.org/en/latest/userguide/workers.html#worker-persistent-revokes ?

Also see the note about using terminate=True : https://docs.celeryproject.org/en/latest/userguide/workers.html#revoke-revoking-tasks 
It should not be used programmatically, but I'm not sure what would be the proper way to stop that task? Maybe using the `signal` parameter with a less destructive signal that could even be catched by the worker and propagate the event to an eventual ADES?

Reply:
I wasn't even aware of persistent revoke. 
This is somewhat bothersome. Why isn't it stored in the same db as tasks 🙄
The fix should be simple enough though with an edit to the Dockerfile. 
It will be up to the server though to mount a directory in the Docker to refer to persistent revokes between restarts.

Regarding `terminate`. I'm not sure why the doc makes so much emphasis on it being as last resort, but directly above it mentions: 
> When a worker receives a revoke request it will skip executing the task, but it won’t terminate an already executing task unless the terminate option is set.

In this case, dismiss operation is supposed to stop a running operation. 
Might need to try with this: https://docs.celeryproject.org/en/latest/reference/celery.contrib.abortable.html
I have not found other alternatives looking at different sources.

As for forwarding the cancellation, in that situation I agree terminate might kill the process before the task can send the request, but at the moment this is not done anyway. I'm leaving it for later with https://github.com/crim-ca/weaver/issues/348

Reply:
> As for forwarding the cancellation, in that situation I agree terminate might kill the process before the task can send the request, but at the moment this is not done anyway. I'm leaving it for later with #348

Yes I agreed that this is not for the current PR scope.